bootstrap/kubeadm/controllers/kubeadmconfig_controller.go

```diff
@@ -563,6 +583,79 @@ func (r *KubeadmConfigReconciler) joinControlplane(ctx context.Context, scope *S
 	return ctrl.Result{}, nil
 }
 
+// resolveProxyUrl adds the user information to the proxy url
+func resolveProxyUrl(proxyUrl string, secret *corev1.Secret) (string, error) {
+	if secret == nil {
+		return proxyUrl, nil
+	}
+	username := secret.StringData["username"]
+	password := secret.StringData["password"]
+	parsedUrl, err := url.Parse(proxyUrl)
+	if err != nil {
+		return "", err
+	}
+	parsedUrl.User = url.UserPassword(username, password)
+	return parsedUrl.String(), nil
+}
+
+// resolveProxyFiles resolves the required proxy unit files to cloudinit.Files
+func (r *KubeadmConfigReconciler) resolveProxyFiles(ctx context.Context, cfg *bootstrapv1.KubeadmConfig) ([]bootstrapv1.File, error) {
```

Comment:
should this be 
```suggestion
func (r *KubeadmConfigReconciler) reconcileProxy(ctx context.Context, cfg *bootstrapv1.KubeadmConfig) ([]bootstrapv1.File, error) {
```
given that there are no proxy files?